pkg/controller/reconciler/albingresscontroller_reconciler.go

```diff
@@ -30,29 +30,29 @@ func (z *albIngressReconciler) SetCommonMetadata(metadata *resourcetree.CommonMe
 }
 
 // Reconcile knows how to do what is necessary to ensure the desired state is achieved
-func (z *albIngressReconciler) Reconcile(node *resourcetree.ResourceNode) (*ReconcilationResult, error) {
+func (z *albIngressReconciler) Reconcile(node *resourcetree.ResourceNode) (result ReconcilationResult, err error) {
```

Comment:
I would be a little bit careful with named return values, there is a section in effective-go on the matter: https://golang.org/doc/effective_go#named-results, where they name some situations where it can be useful:

> The names are not mandatory but they can make code shorter and clearer: they're documentation. If we name the results of nextInt it becomes obvious which returned int is which.

They also come with some problems though, such as variable shadowing: https://medium.com/@Nidhi11594/variable-shadowing-in-golang-f500e8e58931, our linter should pick this up, but beware.

In this situation, I am not sure that returning the empty `ReconciliationResult{}` is that much worse. Your mileage may vary, and to some extent, this is a personal preference. I know what I prefer 😬

Reply:
Yeah, this is my preference coming through. I just find it alot cleaner when I don't have to use `:` (when it makes sense). In the case of named results, I tend to explicitly declare variables to avoid shadowing